pysat/_instrument.py

```diff
@@ -443,7 +443,12 @@ def __getitem_xarray__(self, key):
                 try:
                     return self.data.isel(time=key[0])[key[1]]
                 except:
-                    return self.data.sel(time=key[0])[key[1]]
+                    try:
+                        return self.data.sel(time=key[0])[key[1]]
+                    except TypeError: # construct dataset from names
+                        variables = list(self.data.variables.keys())[key[1]]
+                        dataset_dict = {variable: self.data[variable] for variable in variables if variable not in self.data.coords}
+                        return xr.Dataset(dataset_dict, coords = self.data.coords, attrs = self.data.attrs)
```

Comment:
Can't we return the xarray dataset that already exists in data rather than recreate it?

Reply:
Yes, I think I have it right now. Didn't know you could do `dataset[list]`